server/routerlicious/packages/services-core/src/throttler.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INackContent, NackErrorType } from "@fluidframework/protocol-definitions";
+
+export interface IThrottlerResponse {
+    throttleStatus: boolean;
+    throttleReason: string;
+    retryAfterInMs: number;
+}
+
+export interface IThrottlingMetrics extends IThrottlerResponse {
+    count: number;
+    lastCoolDownAt: number;
+}
+
+export class ThrottlingError implements INackContent {
+    readonly code = 429;
+    readonly type = NackErrorType.ThrottlingError;
+
+    constructor(
+        /**
+         * Explanation for throttling.
+         */
+        readonly message: string,
+        /**
+         * Client should retry operation after this many seconds.
+         */
+        readonly retryAfter: number,
```

Comment:
Better to reflect the unit in the name, retryAfterInSeconds? And I am curious that elsewhere ms is used, why here specifically second is used?

Reply:
The reason for the naming and units here is that `INackContent` is defined and used elsewhere in FFX (e.g. Container Loader) to handle retries. Seemed like the best I could do was the doc comment without drastically increasing the scope of this work